### PR TITLE
Compact messages addon

### PR DESCRIPTION
**Pull request by CodeGuy92**
### Changes

You can check the comments.

### Reason for changes

Because it will make the messages more readable and cool _B)_

### Tests

Capture
This works well.

### CODE REVIEW

**Comment by CodeGuy92** [comment]
i could add an option to customize the padding, but i'm kinda busy rn

**Comment by FunctionalMetatable** [comment]
Can't you use css instead of js?

**Comment by CodeGuy92** [comment]
> Can't you use css instead of js?

i don't know how to do that sry

File: addons/compact-messages/addon.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "Compact Messages",
+  "description": "Makes the messages page compact and easy to read.",
+    "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["https://scratch.mit.edu/messages"]
+    }
+  ],
+  "tags": ["messages", "compact"],
```
Comment (GrahamSH-LLK) [inline addons/compact-messages/addon.json:10]: Tags are not custom, you have to pick one or two from the list.

Comment (CodeGuy92) [reply addons/compact-messages/addon.json:10]: which list

**Comment by CodeGuy92** [comment]
Now you guys can modify the code

File: addons/compact-messages/addon.json (hunk 1 above)
Comment (CST1229) [reply addons/compact-messages/addon.json:10]: [This list](https://scratchaddons.com/docs/reference/addon-manifest/#tags-array-required)

Comment (CodeGuy92) [reply addons/compact-messages/addon.json:10]: i didn't read that

**Comment by GrahamSH-LLK** [comment]
Remove recomended

File: addons/compact-messages/userscript.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// select all of the icons
+var icons = document.getElementsByClassName('social-message-icon');
+// for each icon, execute
+for(var i = 0; i < icons.length; i++){
+  // remove the friken icon
+    icons[i].remove();
+}
+
+// select all of the messages
+var messages = document.getElementsByClassName('social-message');
```
[the remark this answers is not in the document]
Comment (CodeGuy92) [reply addons/compact-messages/userscript.js:15]: Ok, i will do that tomorrow!

**Comment by WorldLanguages** [comment]
@mxmou CSS king, what do you think of this?

Personally, I'd keep the message type icons, and would try lowering the font instead of bringing the padding down so much.

**Comment by CodeGuy92** [comment]
I just tested this locally, works.

**Comment by CodeGuy92** [comment]
> @mxmou CSS king, what do you think of this?
> 
> Personally, I'd keep the message type icons, and would try lowering the font instead of bringing the padding down so much.

To bring in my opinion, do you think i should make it customizable?

**Comment by mxmou** [comment]
> @mxmou CSS king, what do you think of this?
> 
> Personally, I'd keep the message type icons, and would try lowering the font instead of bringing the padding down so much.

I think the padding is fine (@CodeGuy92 but making it customizable would be a good idea). Removing icons should be an option.

Could this addon also work with studio activity and maybe project/studio comments?

**Comment by CodeGuy92** [comment]
Wait bruh how do i import the settings in the userstyle

File: addons/compact-messages/addon.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      "max": 15
+    },
+    {
+      "name": "Comment box padding (5-15 pixels)",
+      "id": "comment_padding",
+      "type": "integer",
+      "default": 5,
+      "max": 15
+    }
+  ]
```
Comment (mxmou) [inline addons/compact-messages/addon.json:25]: Missing comma

**Review by mxmou — CHANGES_REQUESTED** [review]

File: addons/compact-messages/addon.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "Compact Messages",
```
Comment (mxmou) [inline addons/compact-messages/addon.json:2]: We use sentence case in addon names.

File: addons/compact-messages/addon.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  "description": "Makes the messages page compact and easy to read.",
```
Comment (mxmou) [inline addons/compact-messages/addon.json:3]: I think "more compact and easier to read" would be better.

**Comment by mxmou** [comment]
> Wait bruh how do i import the settings in the userstyle

`calc(1px * var(--compactMessages-message_padding))`

For the hide icons setting, move the CSS that hides icons into a separate userstyle and declare it like this:
```json
{
  "url": "hide_icons.css",
  "matches": ["https://scratch.mit.edu/messages"],
  "settingMatch": {
    "id": "hide_icons",
    "value": true
  }
}
```

File: addons/compact-messages/addon.json (hunk 3 above)
Comment (CodeGuy92) [reply addons/compact-messages/addon.json:25]: you mean a comma next to "15"?

File: addons/compact-messages/addon.json (hunk 5 above)
Comment (CodeGuy92) [reply addons/compact-messages/addon.json:3]: ok

**Comment by CodeGuy92** [comment]
> > Wait bruh how do i import the settings in the userstyle
> 
> `calc(1px * var(--compactMessages-message_padding))`
> 
> For the hide icons setting, move the CSS that hides icons into a separate userstyle and declare it like this:
> 
> ```json
> {
>   "url": "hide_icons.css",
>   "matches": ["https://scratch.mit.edu/messages"],
>   "settingMatch": {
>     "id": "hide_icons",
>     "value": true
>   }
> }
> ```

_why_

File: addons/compact-messages/addon.json (hunk 3 above)
Comment (RedGuy12) [reply addons/compact-messages/addon.json:25]:
```suggestion
  ],
```

**Comment by RedGuy12** [comment]
> > For the hide icons setting, move the CSS that hides icons into a separate userstyle and declare it like this:
> > ```json
> > {
> >   "url": "hide_icons.css",
> >   "matches": ["https://scratch.mit.edu/messages"],
> >   "settingMatch": {
> >     "id": "hide_icons",
> >     "value": true
> >   }
> > }
> > ```
> 
> _why_

that's how you do it

the way you're doing it now doesn't work

**Comment by CodeGuy92** [comment]
I've finished everything. I needed to do.

**Comment by CodeGuy92** [comment]
When's this gonna get merged?

**Review by CodeGuy92 — COMMENTED** [review]
*reviews*

File: addons/compact-messages/addon.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      "type": "integer",
+      "default": 5,
+      "max": 15
+    },
+    {
+      "name": "Comment box padding (5-15 pixels)",
+      "id": "comment_padding",
+      "type": "integer",
+      "default": 5,
+      "max": 15,
```
[the remark this answers is not in the document]
Comment (CodeGuy92) [reply addons/compact-messages/addon.json:23]: wdym, it worked.
leme test it again

**Comment by CodeGuy92** [comment]
So i have re-tested locally and checked every line. Now it works perfectly.

image
image
image

File: addons/compact-messages/addon.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (CodeGuy92) [reply addons/compact-messages/addon.json]: check comment below

**Review by CodeGuy92 — COMMENTED** [review]
sooo...

**Comment by CodeGuy92** [comment]
NNOO WAY--- FINALYY

**Comment by CodeGuy92** [comment]
I CONTRIBUTED TO THE REPO

**Comment by CodeGuy92** [comment]
IM HAPPY

**Comment by CodeGuy92** [comment]
Hold on, the limit on both inputs is 16px instead of 15px. That needs to get fixed lol

**Pull request merged**